platforms/solana/src/signer.ts

```diff
@@ -461,6 +472,78 @@ export async function determinePriorityFee(
   return Math.min(Math.max(fee, minPriorityFee), maxPriorityFee);
 }
 
+interface RpcResponse {
+  jsonrpc: String;
+  id?: String;
+  result?: [];
+  error?: any;
+}
+
+// Helper function to calculate the priority fee using the Triton One API
+// See https://docs.triton.one/chains/solana/improved-priority-fees-api
+// NOTE: this is currently an experimental feature
+export async function determinePriorityFeeTritonOne(
+  connection: Connection,
+  transaction: Transaction | VersionedTransaction,
+  percentile: number = DEFAULT_PRIORITY_FEE_PERCENTILE,
+  multiple: number = DEFAULT_PERCENTILE_MULTIPLE,
+  minPriorityFee: number = DEFAULT_MIN_PRIORITY_FEE,
+  maxPriorityFee: number = DEFAULT_MAX_PRIORITY_FEE,
+): Promise<number> {
+  // Start with min fee
+  let fee = minPriorityFee;
+
+  // @ts-ignore
+  const rpcRequest = connection._rpcRequest;
+
+  const accounts = await getWritableAccounts(connection, transaction);
+
+  const args = [
+    accounts,
+    {
+      percentile: percentile * 10_000, // between 1 and 10_000
```

Comment:
i would validate the `percentile` input is a number between 0 and 1, and throw if it's not.

Reply:
good idea